core/src/vaults.ts

```diff
@@ -0,0 +1,378 @@
+        feedEvents = await priceFeedContract.queryFilter(feedEventsFilter, fromBlock, toBlock);
+        rpcEventFilterSegmentIndex += 1;
+        fromBlock = -EVENTS_PER_RPC_REQUEST * (rpcEventFilterSegmentIndex + 1) - EVENTS_ON_FIRST_RPC_REQUEST;
+        toBlock = -EVENTS_PER_RPC_REQUEST * rpcEventFilterSegmentIndex - EVENTS_ON_FIRST_RPC_REQUEST;
+    }
+
+    if (feedEvents.length === 0) {
+        return undefined;
+    }
+
+    const nextUnitCollateralPrice = new BigNumber(feedEvents[feedEvents.length - 1].args?.val._hex).shiftedBy(
+        -WAD_NUMBER_OF_DIGITS
+    );
+
+    return {
+        currentUnitPrice: currentUnitCollateralPrice,
+        nextUnitPrice: nextUnitCollateralPrice,
+        nextPriceChange: new Date((lastPriceUpdateTimestampInSeconds + priceUpdateFrequencyInSeconds) * 1000),
+    };
+};
+
+export const getOsmPrices = memoizee(_getOsmPrices, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+const _fetchLiquidationRatio = async (network: string, collateralType: CollateralType) => {
+    const contract = await getContract(network, 'MCD_SPOT');
+    const collateralTypeAsHex = ethers.utils.formatBytes32String(collateralType);
+    const oracleAndLiquidationRatio = await contract.ilks(collateralTypeAsHex);
+    const liquidationRatioAsHex = oracleAndLiquidationRatio.mat;
+    const oracleAddress = oracleAndLiquidationRatio.pip;
+    const liquidationRatio = new BigNumber(liquidationRatioAsHex._hex).shiftedBy(-RAY_NUMBER_OF_DIGITS).toNumber();
+    return { oracleAddress, liquidationRatio };
+};
+
+export const fetchLiquidationRatio = memoizee(_fetchLiquidationRatio, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+const _fetchVaultLiquidationIncentive = async (
+    network: string,
+    liquidatiorContractAddress: string,
+    debtDai: BigNumber
+) => {
+    const liquidatorContractInterface = await getContractInterfaceByName('MCD_CLIP_');
+    const provider = await getProvider(network);
+    const liquidatorContract = new ethers.Contract(liquidatiorContractAddress, liquidatorContractInterface, provider);
+    const incentiveConstantDaiHex = await liquidatorContract.tip();
+    const incentiveRelativeDaiHex = await liquidatorContract.chip();
+    const incentiveRelativeDai = new BigNumber(incentiveRelativeDaiHex._hex)
+        .shiftedBy(-WAD_NUMBER_OF_DIGITS)
+        .multipliedBy(debtDai);
+    const incentiveConstantDai = new BigNumber(incentiveConstantDaiHex._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS);
+    const incentiveCombinedDai = incentiveRelativeDai.plus(incentiveConstantDai);
+    return {
+        incentiveCombinedDai,
+        incentiveConstantDai,
+        incentiveRelativeDai,
+    };
+};
+
+export const fetchVaultLiquidationIncentive = memoizee(_fetchVaultLiquidationIncentive, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 3,
+});
+
+const _isVaultLiquidated = async (network: string, vault: Vault) => {
```

Comment:
Do liquidated vaults stay within cdp manager and still can be fetched from there?

Anyway, the name of the function is misleading, I would call it `fetchLiquidatedParameters` and return undefined if there is nothing

Reply:
> Do liquidated vaults stay within cdp manager and still can be fetched from there?

yes.